src/main.rs

```diff
@@ -1105,6 +1127,15 @@ mod tests {
         }
     }
 
+    mock! {
+        SendCustomMessenger{}
+
+        #[async_trait]
+         impl SendCustomMessage for SendCustomMessenger{
+             async fn send_custom_message(&mut self, request: SendCustomMessageRequest) -> Result<SendCustomMessageResponse, Status>;
+         }
+    }
+
     #[tokio::test]
     async fn test_consume_messenger_events() {
```

Comment:
I struggled a bit with this because of the while loop where the SendOutgoing event is handled in consume_messenger_events. I tried to do something like... have the mock onion messenger return an onion message the first invocation of `next_onion_message_for_peer`, then the second invocation return None, using this... https://docs.rs/mockall/latest/mockall/#matching-multiple-calls Buuuuut I couldn't figure out how to change the return value when the argument (the peer) is the same each time. So it would just return nonstop onion messages in an infinite while loop. I'll think on this a bit more tomorrow but if you have any ideas LMK.